### PR TITLE
[SWE-34] Create Post Frontend

### PR DESCRIPTION
**Pull request by datderek**
Task ID: [SWE-34](https://linear.app/swe-dem-boys/issue/SWE-34/create-post-frontend)
## PR Description
Add a button to the dashboard that when pressed, displays a modal allowing the user to create a Post. Logs all user inputs entered individually and in the form of an SQL query.

### CODE REVIEW

File: app/components/PostFormModal.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      modalRef.current?.showModal();
+    } else {
+      modalRef.current?.close();
+    }
+  }, [showModal]);
+
+  /**
+   *  Opens the modal
+   */
+  function clickOpen(): void {
```
Comment (jackfales) [inline app/components/PostFormModal.tsx:20]: Replace with [arrow function](https://www.tutorialsteacher.com/typescript/arrow-function)

File: app/components/PostFormModal.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    setShowModal(true);
+  }
+
+  /**
+   * Closes the modal
+   */
+  function clickClose(): void {
```
Comment (jackfales) [inline app/components/PostFormModal.tsx:27]: Same as above

File: app/components/PostFormModal.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    setShowModal(false);
+  }
+
+  /**
+   * Submits new post data then closes the modal
+   * 
+   * @param e - the form submit event
+  */
+  function clickOk(e: FormEvent<HTMLFormElement>): void {
```
Comment (jackfales) [inline app/components/PostFormModal.tsx:36]: Rename to something more intuitive. `clickOk` doesn't accurately describe what the function is used for. Maybe `onSubmitHandler` or something similar?

File: app/components/PostFormModal.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    e.preventDefault();
+    const formData: Object = Object.fromEntries((new FormData(e.currentTarget)));
+    console.log(JSON.stringify(formData));
+    
+    const sql: String = `INSERT INTO posts (name, image_url, caption, recipe, tags)
+    VALUES (${formData['name']}, /hardcodedimageurl, ${formData['caption']},
+    ${formData['recipe']}, ${formData['tags']}})`;
```
Comment (jackfales) [inline app/components/PostFormModal.tsx:43]: For now this is fine but this is vulnerable to [SQL injection attacks ](https://www.w3schools.com/sql/sql_injection.asp)

**Review by jackfales — APPROVED** [review]

**Review by sammyoto — APPROVED** [review]
Beautiful. Looks good to me.

**Pull request merged**